### ISSUE TITLE
Support refactor/rename

### ISSUE THREAD

**Issue by jan-xyz** [aznhe21/actions-preview.nvim#31]
Have you thought about also supporting refactor/rename? It supports the same mechanism as code actions and it would be really nice to also have it!

**Comment by aznhe21** [comment]
Good idea! The implementation doesn't seem particularly difficult either. However, it will take time to get the overall balance and UI right. Please wait a while.

**Comment by pidgeon777** [comment]
I confirm it would be simply great.

It would give us a chance to preview the effect of the LSP rename/refactor action before actually applying it.

This mean for example:
- Replacements in all of the involved files (opened or not)
- Files rename

and so on, according to the LSP capabilities.

Finally, I think that actually there are no plugins supporting such a feature, so yours would be the first one to implement it.

**Comment by jan-xyz** [comment]
@pidgeon777 I spend some time on a similar plugin, and I would be interested to get your feedback, as it also implements `refactor/rename`: https://github.com/jan-xyz/lsp-preview.nvim

**Comment by pidgeon777** [comment]
I just gave a look at your plugin and it seems to be a great one. Congratulations for the work 👍.

I'll open specific issues in your repository to discuss about any improvements or found bugs.
